src/Castle.Windsor/Core/LifestyleType.cs

```diff
@@ -46,13 +46,6 @@ public enum LifestyleType
 		/// </summary>
 		Pooled,
 
-#if FEATURE_SYSTEM_WEB
-		/// <summary>
-		///   PerWebRequest components are created once per Http Request
-		/// </summary>
-		PerWebRequest,
```

Comment:
Can you please clarify? 

I would really like to make this enum interal and hide it all together if I could. Future developers will naturally gravitate toward trying to extend this for lifestyles. What we are really saying is your intent should be established via the component registration without this enum. Also Windsor is choc full when it comes to lifestyles. Implementing platform/framework lifestyle should be done using scoping.

Also I have done all the work to remove it all together from the underlying components. So this is my mind is an intentional breaking change.

Reply:
> I would really like to make this enum interal and hide it all together if I could. Future developers will naturally gravitate toward trying to extend this for lifestyles. What we are really saying is your intent should be established via the component registration without this enum. Also Windsor is choc full when it comes to lifestyles. Implementing platform/framework lifestyle should be done using scoping.

Over time we'll push towards getting lifetimes out of here, however for at least singleton and transient there would be so much code using this enum I'd vote against removing this from the public API even if guidance was to use the extension methods.

Here I was just asking that we change the LifestyleType enum members so they have explicit values to avoid an unnecessary breaking change for Custom, Scoped and Bound changing integer values:

	public enum LifestyleType
	{
		Undefined = 0,
		Singleton = 1,
		Thread = 2,
		Transient = 3,
		Pooled = 4,
		// 5 gets left unassigned
		Custom = 6,
		Scoped = 7,
		Bound = 8
	}

Reply:
Ok, got it. Done.

Reply:
Addressed in commit:  32ab15e

Reply:
PS: I prefer the iota in [golang](https://golang.org/ref/spec#Iota).